### PR TITLE
feat(async/unstable): accept iterator varargs in `MuxAsyncIterator`

### LINKED ISSUE #6086: @std/async: make `MuxAsyncIterator` accept varargs
**Is your feature request related to a problem? Please describe.**

```ts
const mux = new MuxAsyncIterator<number>()
mux.add(gen123())
mux.add(gen456())
```

- current method is too imperative to be practical.
- type cannot be inferred from iterators added.

**Describe the solution you'd like**

```ts
const mux = new MuxAsyncIterator(gen123(), gen456())
```

- make `MuxAsyncIterator` accept variable.
- it has an added benefit of type inference via constructor.

### PR DESCRIPTION
**Pull request by scarf005**
- resolves #6086

### CODE REVIEW

**Comment by kt3k** [comment]
Thanks for the PR.

> not sure how to handle unstable as it needs to modify class constructor.

I think we can extend the class and define the constructor in the subclass in `unstable_mux_async_iterator.ts`. Can you work on that change?

**Comment by scarf005** [comment]
> Thanks for the PR.
> 
> > not sure how to handle unstable as it needs to modify class constructor.
> 
> I think we can extend the class and define the constructor in the subclass in `unstable_mux_async_iterator.ts`. Can you work on that change?

applied in [`7a7efd2` (#6087)](https://github.com/denoland/std/pull/6087/commits/7a7efd23724898a1fef29c4f95ddfd0b50d990c9).

**Comment by kt3k** [comment]
I changed the implementation a bit to reduce the repetation.

**Review by kt3k — APPROVED** [review]
LGTM

File: async/unstable_mux_async_iterator.ts [hunk 1]
```diff
@@ -40,134 +37,14 @@ interface TaggedYieldedValue<T> {
...
  * @typeParam T The type of the provided async iterables and generated async iterable.
  */
-export class MuxAsyncIterator<T> implements AsyncIterable<T> {
-  #iteratorCount = 0;
-  #yields: Array<TaggedYieldedValue<T>> = [];
-  // deno-lint-ignore no-explicit-any
-  #throws: any[] = [];
-  #signal = Promise.withResolvers<void>();
-
+export class MuxAsyncIterator<T> extends MaxAsyncIterator_<T> {
```
Comment (scarf005) [inline async/unstable_mux_async_iterator.ts:40]: `MaxAsyncIterator_` this seems to be a typo of `MuxAsyncIterator_`?

Comment (kt3k) [reply async/unstable_mux_async_iterator.ts:40]: Ah, true. Thanks for pointing. Can you send a PR?

Comment (scarf005) [reply async/unstable_mux_async_iterator.ts:40]: opened #6096

**Pull request merged**